src/operator/nn/dropout-inl.h

```diff
@@ -130,11 +130,18 @@ class DropoutOp {
     DType *dataptr = data.dptr_;
     auto maskptr = reinterpret_cast<int *>(mask.dptr_);
     int count = mask.shape_[0] * mask.shape_[1];
+    if (sizeof(DType) > sizeof(int)) {
+      // allocating new buffer to avoiding memory overlapping between `mask.dptr_` and `maskptr`
+      Tensor<xpu, 1, int> temp = ctx.requested[1].get_space_typed<xpu, 1, int>(Shape1(count), s);
+      maskptr = temp.dptr_;
```

Comment:
will temp be deleted when out of if section?

Reply:
Seems will not be deleted. I put it here mainly because `mask` buffer can not be reused when `sizeof(DType) > sizeof(int)` and must allocate new memory space to avoid memory overlapping.